exercises/house/house_test.go

```diff
@@ -1,19 +1,13 @@
-// Embed places a phrase with another phrase.
-//
-//    func Embed(prefixPhrase, suffixPhrase string) string
-
-// Verse generates a verse with relative phrases that have a recursive structure.
-//
-//    func Verse(prefixPhrase string, relPhrases []string, suffixPhrase string) string
-
-// As ever, there are different ways to do this. Try using Embed as a
-// subroutine of Verse, and using programmatic recursion to generate the return
-// value, reflecting the grammatical recursion of the song.
+// As ever, there are different ways to complete this exercise.
+// Try using using programmatic recursion to generate the verses of the song,
+// thus reflecting the song's grammatical recursion.
 
-// Song generates the full text of "The House That Jack Built". Of course, you
-// could return a string literal, but humor us; try using Verse as a subroutine.
+// While recursion isn't always the simplest or most efficient solution to a problem,
+// it's a powerful programming technique nonetheless.
+// New to recursion? Here are some useful articles to read up on it:
 //
-//    func Song() string
+// https://www.golang-book.com/books/intro/7#section5
+// https://www.goinggo.net/2013/09/recursion-and-tail-calls-in-go_26.html
```

Comment:
this one is a lot to absorb for someone new to recursion (the supposed target audience for these links). I say this because it describes details such as the stack layout, the difference between a recursive call in tail-call position versus not, and how the compiler might treat each.

The [image showing the call stack](https://www.goinggo.net/images/goinggo/Screen+Shot+2013-09-20+at+6.36.41+PM.png) was useful. I would argue that the rest of the page may not appeal to someone new to recursion. I can't say for sure ([curse of knowledge](https://en.wikipedia.org/wiki/Curse_of_knowledge), etc.) but I was guessing that such a student would be interested simply in "what is recursion? how do I use it? how can I reason through what happens with a recursive function, so that I know how to code one?" and not how to optimise one (tail-call versus not).

What say you? Maybe I'm wrong, and people are interested!